pylint/checkers/base.py

```diff
@@ -512,13 +512,29 @@ def visit_functiondef(self, node):
                            retnode.value.value is not None:
                         self.add_message('return-arg-in-generator', node=node,
                                          line=retnode.fromlineno)
-        # Check for duplicate names
-        args = set()
-        for name in node.argnames():
-            if name in args:
-                self.add_message('duplicate-argument-name', node=node, args=(name,))
+        # Check for duplicate names by clustering args with same name for detailed report
+        args = node.args
+        arg_clusters = collections.defaultdict(list)
+        for arg in itertools.chain.from_iterable(pack for pack in
+                                                 [args.args, [args.vararg],
+                                                  args.kwonlyargs, [args.kwarg]]
+                                                 if pack is not None and all(pack)):
+            if isinstance(arg, str):
+                # sadly, asteroid doesn't expose variadic and kw-variadic arg nodes
+                arg_clusters[arg].append(arg)
             else:
-                args.add(name)
+                arg_clusters[arg.name].append(arg)
+
+        # provide detailed report about each repeated argument
+        for arg in itertools.chain.from_iterable(args for args in arg_clusters.values()
+                                                 if len(args) > 1):
+            if isinstance(arg, str):
+                # fallback on previous behavior
+                add_message_args = node.lineno, node, (arg,)
```

Comment:
This inconsistency bothers me a little bit - if we cannot _always_ emit an error pinpointed to given argument, I don't see much value in merging this change at all.

Reply:
Sure, i sad about this too. Probably, this is astroid issue. 
But i think it's better to provide as much as possible helpful information for user, even when there is some inconsistency.

Reply:
Agree with @rogalski here, as mentioned in my previous comment, we can just ignore variadic params for now.